software/teachee-desktop/src/app.rs

```diff
@@ -101,7 +113,11 @@ impl eframe::App for App {
                 ui.visuals_mut().button_frame = false;
                 widgets::global_dark_light_mode_switch(ui);
                 ui.separator();
-                ui.label(if *flag { "Flag: 1" } else { "Flag: 0" });
+                ui.label(if storage.lock().unwrap().read_flag() {
```

Comment:
I think we had different ideas for what change you were planning to make. I thought that you were going to move `spawn_usb_manager` to it's own struct so we could add fields to it later. This would allow for USB manager to have space to write bookkeeping information while Storage is locked.

Given that `update` should do almost no heavy computation and is called at a rate of only 60Hz, I don't think the old implementation which held the lock for the entire function was problematic. This new approach will lead to us repeatedly acquiring and releasing the lock which

1. Adds a lot of noise to the code.
2. Maybe adds some overhead vs a single lock? Not sure though.

I would prefer sticking with the existing implementation until we run into performance issues.

Reply:
Still not sure what you intended `Storage` be used for. Are you saying to move `Arc<Mutex<App>>` to a new field in a usb manager struct? Why does the App have to be associated with the usb manager? I would prefer that GUI and usb functionality be completely separate

> repeatedly acquiring and releasing the lock

We can acquire at function start if needed.

Reply:
But I was also thinking of adding a struct for usb manager

Reply:
> Still not sure what you intended Storage be used for.

Sorry, let me clarify. The entrypoint for `eframe` requires we return a type which implements `eframe::App`. We also want this type to be thread safe so we edit UI state from the USB manager. Rust only lets you implement traits on your own type so `Storages` is created as a workaround. We are only using it as `Arc<Mutex<App>>` we can implement traits for.

Reply:
What UI state needs to be edited from the usb thread? Is it not enough for the UI thread to manage UI? We should defer as much work as possible to the UI thread since it's only executing at 60 Hz as you said